mtxclient, nheko fail to build on aarch64-linux

### Steps To Reproduce
Steps to reproduce the behavior:
1. `nix build '.#legacyPackages.aarch64-linux.nheko'`

### Build log
```
error: builder for '/nix/store/x2p0k768qavk91cwq9jv4xwcgpyhrscm-mtxclient-0.9.0.drv' failed with exit code 2;
       last 10 log lines:
       > make[2]: *** [CMakeFiles/matrix_client.dir/build.make:370: CMakeFiles/matrix_client.dir/lib/structs/events/encrypted.cpp.o] Error 1
       > In file included from /build/source/include/mtxclient/crypto/client.hpp:17,
       >                  from /build/source/lib/crypto/utils.cpp:17:
       > /build/source/include/mtx/identifiers.hpp:12:10: fatal error: compare: No such file or directory
       >    12 | #include <compare>
       >       |          ^~~~~~~~~
       > compilation terminated.
       > make[2]: *** [CMakeFiles/matrix_client.dir/build.make:132: CMakeFiles/matrix_client.dir/lib/crypto/utils.cpp.o] Error 1
       > make[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/matrix_client.dir/all] Error 2
       > make: *** [Makefile:136: all] Error 2
       For full logs, run 'nix log /nix/store/x2p0k768qavk91cwq9jv4xwcgpyhrscm-mtxclient-0.9.0.drv'.
error: 1 dependencies of derivation '/nix/store/szdzykpfiknzmzl5j0mar4kvngnq7hig-nheko-0.11.1.drv' failed to build
```

### Additional context
- i noticed the break after the nheko 0.10.2 -> 0.11.0 upgrade landed in nixos-unstable: <https://github.com/NixOS/nixpkgs/pull/210471>
- hydra failure (click `log` for the build log): <https://hydra.nixos.org/build/205928297>
- build is currently broken in both master (5ed481943351e9fd354aeb557679624224de38d5) and ~~nixos-22.11 branches~~has been fixed in nixos-22.11, see https://github.com/NixOS/nixpkgs/pull/211776#issuecomment-1399363381

### Notify maintainers

- tagging @rnhmjoj for having initiated the backport.
- tagging @fpletz as maintainer of both mtxclient and nheko. 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[colin@uninsane.org:~/dev/3rd/nixpkgs]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.86, NixOS, 23.05 (Stoat), 23.05pre-git`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.12.0`
 - nixpkgs: `/home/colin/dev/3rd/nixpkgs`
```

--- a/pkgs/applications/networking/instant-messengers/nheko/default.nix
+++ b/pkgs/applications/networking/instant-messengers/nheko/default.nix
@@ -1,6 +1,5 @@
 { lib
 , stdenv
-, mkDerivation
 , fetchFromGitHub
 , cmake
 , asciidoc
@@ -26,12 +25,13 @@
 , qttools
 , re2
 , spdlog
+, wrapQtAppsHook
 , voipSupport ? true
 , gst_all_1
 , libnice
 }:
 
-mkDerivation rec {
+stdenv.mkDerivation rec {
   pname = "nheko";
   version = "0.11.1";
 
@@ -47,6 +47,7 @@ mkDerivation rec {
     cmake
     lmdbxx
     pkg-config
+    wrapQtAppsHook
   ];
 
   buildInputs = [
@@ -81,6 +82,9 @@ mkDerivation rec {
   cmakeFlags = [
     "-DCOMPILE_QML=ON" # see https://github.com/Nheko-Reborn/nheko/issues/389
   ];
+
+  # https://github.com/NixOS/nixpkgs/issues/201254
+  NIX_LDFLAGS = lib.optionalString (stdenv.isLinux && stdenv.isAarch64 && stdenv.cc.isGNU) "-lgcc";
 
   preFixup = lib.optionalString voipSupport ''
     # add gstreamer plugins path to the wrapper

--- a/pkgs/development/libraries/mtxclient/default.nix
+++ b/pkgs/development/libraries/mtxclient/default.nix
@@ -53,6 +53,9 @@ stdenv.mkDerivation rec {
     spdlog
   ];
 
+  # https://github.com/NixOS/nixpkgs/issues/201254
+  NIX_LDFLAGS = lib.optionalString (stdenv.isLinux && stdenv.isAarch64 && stdenv.cc.isGNU) "-lgcc";
+
   meta = with lib; {
     description = "Client API library for the Matrix protocol.";
     homepage = "https://github.com/Nheko-Reborn/mtxclient";

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -31546,7 +31546,10 @@ with pkgs;
 
   ngt = callPackage ../development/libraries/ngt { };
 
-  nheko = libsForQt5.callPackage ../applications/networking/instant-messengers/nheko { };
+  nheko = libsForQt5.callPackage ../applications/networking/instant-messengers/nheko {
+    # https://github.com/NixOS/nixpkgs/issues/201254
+    stdenv = if stdenv.isLinux && stdenv.isAarch64 && stdenv.cc.isGNU then gcc11Stdenv else stdenv;
+  };
 
   nomacs = libsForQt5.callPackage ../applications/graphics/nomacs { };
 

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -22021,7 +22021,10 @@ with pkgs;
 
   mtpfs = callPackage ../tools/filesystems/mtpfs { };
 
-  mtxclient = callPackage ../development/libraries/mtxclient { };
+  mtxclient = callPackage ../development/libraries/mtxclient {
+    # https://github.com/NixOS/nixpkgs/issues/201254
+    stdenv = if stdenv.isLinux && stdenv.isAarch64 && stdenv.cc.isGNU then gcc11Stdenv else stdenv;
+  };
 
   mu = callPackage ../tools/networking/mu {
     texinfo = texinfo4;

